components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionSecretProcessor.java

```diff
@@ -0,0 +1,192 @@
+                decryptedAuthProperties.add(new AuthProperty.AuthPropertyBuilder()
+                        .name(prop.getName())
+                        .isConfidential(prop.getIsConfidential())
+                        .value(resolvedSecret.getResolvedSecretValue()).build());
+            }
+        }
+        return decryptedAuthProperties;
+    }
+
+    public void deleteAssociatedSecrets(List<AuthProperty> authProperties, String actionId, String authType)
+            throws SecretManagementException {
+
+        for (AuthProperty prop : authProperties) {
+            if (!prop.getIsConfidential()) {
+                continue;
+            }
+            String secretName = buildSecretName(actionId, authType, prop.getName());
+            if (ActionMgtServiceComponentHolder.getInstance().getSecretManager()
+                    .isSecretExist(IDN_SECRET_TYPE_ACTION_SECRETS, secretName)) {
+                ActionMgtServiceComponentHolder.getInstance().getSecretManager()
+                        .deleteSecret(IDN_SECRET_TYPE_ACTION_SECRETS, secretName);
+            }
+        }
+    }
+
+    public List<AuthProperty> getPropertiesWithSecretReferences(List<AuthProperty> authProperties, String actionId,
+                                                                String authType) throws SecretManagementException {
+
+        List<AuthProperty> referenceUpdatedProperties = new ArrayList<>();
+        for (AuthProperty prop : authProperties) {
+            if (!prop.getIsConfidential()) {
+                referenceUpdatedProperties.add(prop);
+                continue;
+            }
+            referenceUpdatedProperties.add(new AuthProperty.AuthPropertyBuilder()
+                    .name(prop.getName())
+                    .isConfidential(prop.getIsConfidential())
+                    .value(buildSecretReference(buildSecretName(actionId, authType, prop.getName()))).build());
+        }
+        return referenceUpdatedProperties;
+    }
+
+    /**
+     * Create secret name.
+     *
+     * @param actionId     Action Id.
+     * @param authType     Authentication Type.
+     * @param authProperty Authentication Property.
+     * @return Secret Name.
+     */
+    private String buildSecretName(String actionId, String authType, String authProperty) {
+
+        return actionId + ":" + authType + ":" + authProperty;
+    }
+
+    /**
+     * Create secret reference name.
+     *
+     * @param secretName Name of the secret.
+     * @return Secret reference name.
+     * @throws SecretManagementException If an error occurs while retrieving the secret type.
+     */
+    private String buildSecretReference(String secretName) throws SecretManagementException {
+
+        if (secretType == null) {
+            secretType = ActionMgtServiceComponentHolder.getInstance().getSecretManager()
+                    .getSecretType(IDN_SECRET_TYPE_ACTION_SECRETS);
+        }
```

Comment:
You can instantiate this at the class constructor right

Reply:
Or as per the pattern you have followed rather than having an instance variable you can directly invoke the secret manager and get the secret type

Reply:
FIxed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/4c40292ff2e65531731f8e68897c74215dae5279